### PR TITLE
Use aws-for-fluent-bit `init` image and add ECS Metadata

### PR DESCRIPTION
**Pull request by AshCorr**
## What does this change?

Use `aws-for-fluent-bit`s `init` variant instead of `latest`. The `init` variant has a bootstrap script that injects ECS metadata & makes using multiple custom config files easier.

https://github.com/aws/aws-for-fluent-bit/blob/develop/init/fluent_bit_init_process.go
https://github.com/aws/aws-for-fluent-bit/blob/mainline/use_cases/init-process-for-fluent-bit/README.md

This will be a breaking change so ideally https://github.com/guardian/devx-logs/pull/33 would go down first and allow us to publish a new major version!

### CODE REVIEW

File: ecs/Dockerfile [hunk 1]
```diff
@@ -1,3 +1,11 @@
...
-ADD parsers.conf /parsers.conf
-ADD custom.conf /custom.conf
+FROM amazon/aws-for-fluent-bit:init-2.32.0
+
+ENV aws_fluent_bit_init_file_1 /guardian.conf
+ENV aws_fluent_bit_init_file_2 /guardian-parsers.conf
+
+# Added by aws-for-fluent-bit base image
+# https://github.com/aws/aws-for-fluent-bit/blob/develop/scripts/dockerfiles/Dockerfile.main-release#L49
+ENV aws_fluent_bit_init_file_3 /ecs/ecs-metadata.conf
```
Comment (AshCorr) [inline ecs/Dockerfile:8]: I'm a little bit unsure whether we actually need to include ecs-metadata or if its included by default 🤔

File: ecs/guardian.conf [hunk 2]
```diff
@@ -1,6 +1,3 @@
-[SERVICE]
```
Comment (AshCorr) [inline ecs/guardian.conf:1]: Believe we no longer needs to specify a Service. the init process will recognize guardian-parsers.conf as a parsers file and create a service for us.

File: ecs/guardian-parsers.conf
Comment (AshCorr) [inline ecs/guardian-parsers.conf:1]: Renamed to make it clearer that these files have been added by us and not part of the base image.

Comment (akash1810) [reply ecs/guardian-parsers.conf:1]: We'll need to change the [`configFileValue` downstream](https://github.com/guardian/service-catalogue/blob/bf768ac98a99158bd83876060148a5e247f68995/packages/cdk/lib/ecs/task.ts#L193) too.

**Review by akash1810 — APPROVED** [review]
Brilliant!!

File: ecs/guardian-parsers.conf
Comment (AshCorr) [reply ecs/guardian-parsers.conf:1]: > We'll need to change the [`configFileValue` downstream](https://github.com/guardian/service-catalogue/blob/bf768ac98a99158bd83876060148a5e247f68995/packages/cdk/lib/ecs/task.ts#L193) too.

We should be able to get rid of configFileValue, the init image uses environment variables to decide which config files to load.

**Pull request merged**